java/fury-core/src/main/java/org/apache/fury/serializer/collection/ImmutableCollectionSerializers.java

```diff
@@ -105,6 +105,7 @@ class MapNStub {}
   public static class ImmutableListSerializer extends CollectionSerializer {
     public ImmutableListSerializer(Fury fury, Class cls) {
       super(fury, cls, true);
+      setImmutable(true);
```

Comment:
If value is not immutable, we should not mark this list as immutable. Maybe we need to check value types in copy implementation

Reply:
Yes, you are right, value maybe mutable. We should iterate over the elements in `collection / map`, call `fury.copy()` method and then add them to the new object.